src/NodaTime/Calendars/RegularWeekYearRule.cs

```diff
@@ -124,7 +175,19 @@ public LocalDate GetLocalDate(int weekYear, int weekOfWeekYear, IsoDayOfWeek day
                     throw new ArgumentOutOfRangeException(nameof(weekYear),
                         $"The combination of {nameof(weekYear)}, {nameof(weekOfWeekYear)} and {nameof(dayOfWeek)} is invalid");
                 }
-                return new LocalDate(yearMonthDayCalculator.GetYearMonthDay(days).WithCalendar(calendar));
+                LocalDate ret = new LocalDate(yearMonthDayCalculator.GetYearMonthDay(days).WithCalendar(calendar));
+                // The requested week year and the actual calendar year don't match, it could be due to short weeks
+                // in a non-regular rule. The simplest way to find out is just to check what the week year is...
+                // This problem doesn't apply to rules with regular weeks.
```

Comment:
That's a bit confusing: it's not a 'problem' at all in the ISO case: it's a valid state.

Perhaps reword as "For rules with irregular weeks, the requested week year and actual calendar year should always match. If they don't, the caller has specified an invalid combination (like the Thursday in week 2016-01, which doesn't exist)."

(I also wonder if we should invert the sense of the boolean, or name it more specifically as e.g. `actuallyUseBclShortWeeks` - or something :).)